src/components/Notifications/EventLog/EventLogTreeFilter.tsx

```diff
@@ -0,0 +1,214 @@
+                chips: parentFilter.children?.filter(childNode => !!childNode.checkProps.checked).map(childFilter => ({
+                    name: childFilter.name as string,
+                    value: childFilter.id,
+                    isRead: true
+                })) ||
+                [
+                    {
+                        name: parentFilter.name,
+                        value: parentFilter.id,
+                        isRead: true
+                    }
+                ]
+            } as EventLogCustomFilter));
+
+            const areEqual = newCustomFilters.every((entry, idx) => {
+                if (!!prev[idx]) {
+                    const prevChips = prev[idx].chips.map(chip => chip.value);
+                    const currChips = entry.chips.map(chip => chip.value);
+                    if (entry.bundleId === prev[idx].bundleId &&
+                        entry.chips.every(chip => prevChips.includes(chip.value)) &&
+                        prev[idx].chips.every(chip => currChips.includes(chip.value))
+                    ) {
+                        return true;
+                    }
+                }
+
+                return false;
+            });
+
+            return areEqual ? prev : newCustomFilters;
+        }));
+    }, [ groups, flattenTree, updateFilters ]);
+
+    const onCheck = (event: ChangeEvent<Element>, treeNode: TreeNodeItem, parentNode: TreeNodeItem) => {
+        const checked = (event.target as HTMLInputElement).checked;
+        setTreeNodeById(produce((prev) => {
+            if (!!parentNode) {
+                const children = prev[parentNode.id].children;
+                children?.some(childNode => {
+                    if (childNode.id === treeNode.id) {
+                        childNode.checkProps.checked = checked;
+                        return true;
+                    }
+
+                    return false;
+                });
+
+                if (allChildrenChecked(prev[parentNode.id])) {
+                    prev[parentNode.id].checkProps.checked = true;
+                }
+                else if (childChecked(prev[parentNode.id])) {
+                    prev[parentNode.id].checkProps.checked = null;
+                }
+                else {
+                    prev[parentNode.id].checkProps.checked = checked;
+                }
+            }
+            else {
+                prev[treeNode.id].checkProps.checked = checked;
+                prev[treeNode.id].children?.forEach(leafNode => leafNode.checkProps.checked = checked);
+            }
+        }));
+    };
+
+    return (
+        <Dropdown
+            toggle={ <DropdownToggle
+                onToggle={ () => setIsToggled(!isToggled) }
+                toggleIndicator={ AngleDownIcon }
+            >
+                {placeholder}
+            </DropdownToggle> }
+            isOpen={ isToggled }
+        >
+            <TreeView data={ treeDataArray } hasChecks={ true } onCheck={ onCheck as any } />
```

Comment:
What's the complain of the `onCheck`  (i.e. why the cast to `any`) ?

Reply:
I use my own typing for the params of the onCheck, so it is complaining because the function param types are different